starlite/__init__.py

```diff
@@ -121,3 +126,17 @@
     "route",
     "websocket",
 ]
+
+_dynamic_imports = {"TestClient", "create_test_client", "create_test_request"}
+
+
+# pylint: disable=import-outside-toplevel
+def __getattr__(name: str) -> Any:
+    """Provide lazy importing as per https://peps.python.org/pep-0562/"""
```

Comment:
interesting, I have never seen this used before. 

So this will prevent a breaking change by keeping the import from starlite. The problem is how to prevent users from continuing to import from starlite in the future - I assume the IDE will not try to import from here by default, correct?

Reply:
At first, I also raised deprecation warning, as well as removed these names from `__all__`. But in https://github.com/starlite-api/starlite/pull/174/commits/c81d99a98a09b567641ee682ecf51888336f541a I thought, why forbid it if we can handle it gracefully anyway?

Currently, IDE will still import it from starlite package, since it's declared in `__all__` and imported in `if TYPE_CHECKING` block — so from IDE's perspective not much has changed.

If we decide that we want to discourage it though, then removing the names from `__all__` and `if TYPE_CHECKING` block would be sufficient for redirecting auto-import to `starlite.testing`.

Reply:
Ok, Im not 100% sure I like this, perhaps deprecations and discouragement are better -- @peterschutt whats your opinion here?